x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/199_query_rulesets_before_setup.yml

```diff
@@ -0,0 +1,26 @@
+setup:
+    - skip:
+          version: " - 8.9.99"
+          reason: Introduced in 8.10.0
+
+---
+"Get query ruleset returns a 404 when no query rulesets exist":
+  - do:
+      catch: "missing"
+      query_ruleset.get:
+        ruleset_id: test-i-dont-exist
+
+---
+"Delete query ruleset returns a 404 when no query rulesets exist":
+  - do:
+      catch: "missing"
```

Comment:
`catch: "missing"` will also match an index not found error (discovered when I was testing this for my own bug 😅 ).

Catching the message like this will ensure we're catching the right error:

`catch: /query rule not found/ `

Reply:
Good call, I'll update